python/cudf/cudf/core/buffer.py

```diff
@@ -102,8 +108,8 @@ def _init_from_array_like(self, data, owner):
                 f"Cannot construct Buffer from {data.__class__.__name__}"
             )
 
-    def serialize(self):
-        header = {}
+    def serialize(self) -> Tuple[dict, list]:
+        header = {}  # type: Dict[Any, Any]
```

Comment:
Are we able to merge the declaration with the definition to eliminate the explicit type?
```suggestion
    def serialize(self) -> Tuple[dict, list]:
        header = {
          "type-serialized": pickle.dumps(type(self)),
          "constructor-kwargs": {},
          "desc": self.__cuda_array_interface__.copy()
        }
```

Reply:
* It's possible in this particular instance of `serialize` but not always. For example [here](https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/core/column/column.py#L1184-L1200) it's not trivial to construct `header` with a one-liner.

* If we did do this, MyPy will infer the type of the dictionary, and that won't necessarily be `Dict[Any, Any]`. In fact, here it is inferred as `Dict[str, Any]`, which will preclude later adding non-string keys.

Reply:
Further question: would that be a good thing to do?

Reply:
I'd say explicit types here are better than implicit